src/net.cpp

```diff
@@ -193,30 +193,30 @@ bool IsPeerAddrLocalGood(CNode *pnode)
            IsReachable(addrLocal.GetNetwork());
 }
 
-// pushes our own address to a peer
 void AdvertiseLocal(CNode *pnode)
 {
-    if (fListen && pnode->fSuccessfullyConnected)
+    assert(fListen);
+    // First consider our "best" addr for the peer, as seen locally.
```

Comment:
```suggestion
    // In absence of other information, we assume that the locally seen address is the "best" addr for peers to refer to our node.
```

Reply:
I am not sure whether I am understanding this right. Is this suggesting that we keep track of each `addr` that some peer refers to us as separately, or are we catering to the possibility that our IP from remote looks different than what we see locally?

Reply:
>we keep track of each addr that some peer refers to us as separately, or are we catering to the possibility that our IP from remote looks different than what we see locally?

Sort of both :)
A remote peer always tells us via which IP they see us (within VERSION message, see `addrMe` field there). Then, we store this to consider in future. Among all these options, we sometimes choose the best to advertise it to a remote peer instead of advertising what *we think* is our local addr.

Reply:
@naumenkogs Was it on purpose that you left the original comment in while also adding the suggestion? Just checking...

Reply:
Yes it was. It still applies, I'm just adding a bit clarification I hope :)